test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -193,8 +193,10 @@ public void tearDown() throws Exception {
 		for (String directory : dirs) {
 			File fdir = new File("." + File.separator + directory);
 			File[] files = fdir.listFiles();
-			for (File file : files) {
-				file.delete();
+			if (files != null) {
```

Comment:
This still makes assumptions that there are no subdirectories or other non-file entries, but it is an improvement.